src/AutoRest.CSharp/Common/AutoRest/Plugins/Configuration.cs

```diff
@@ -95,7 +95,10 @@ public enum UnreferencedTypesHandlingOption
                 _absoluteProjectFolder = Path.GetFullPath(Path.Combine(outputFolder, projectFolder));
             }
 
-            if (publicClients && generation1ConvenienceClient)
+            var isAzureProject = ns != null && (ns.StartsWith("Azure.") || ns.StartsWith("Microsoft.Azure"));
```

Comment:
This check skips the validation when the namespace is not set. Should we do this?

Reply:
In our build context we assign library name to the namespace if its not set don't we and library name has to always be set?

If that is the case perhaps we promote that default logic into configuration and remove it from buildcontext.